src/Microsoft.AspNetCore.Server.IntegrationTesting/PublishedApplication.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting
+{
+    public class PublishedApplication: IDisposable
+    {
+        public string Path { get; }
+
+        public PublishedApplication(string path)
+        {
+            Path = path;
+        }
+
+        public void Dispose()
+        {
+            RetryHelper.RetryOperation(() => Directory.Delete(Path, true), _ => { });
```

Comment:
You lost the failure log

Reply:
Yeah, I was considering adding Dispose overload with Logger

Reply:
Or passing the logger into the ctor.

Reply:
That's harder in cached publisher case.

Reply:
I lied, it's easy. Done.